### PR TITLE
nixos/davical: init

### PR DESCRIPTION
**Pull request by adenseaman**
###### Motivation for this change

Adding a more robust CalDAV and CardDAV option to NixOS ecosystem

###### Things done

- [x] Tested using sandboxing (in NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

* This is a web app so it doesn't have any binaries.  But I built it into a NixOS container and verified that it works properly by using its web service to create a new user and uploaded thousands of calendar and address book items to it using vdirsync.  It also passes the NixOS test I made for it.
* It has an initialization service that automatically initializes the necessary database on first install and updates it when a new version is detected.  I've tested this service for correct operation, too.
* DAViCal closure is 44 MB in size according to "nix path-info -S"

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (evils) [inline maintainers/maintainer-list.nix:3954]: additions to maintainer-list.nix should be separate commits

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121406` at 8d040351 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>davical</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/servers/web-apps/davical/default.nix:33:5:
```
   |
33 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/servers/web-apps/davical/default.nix:20:3:
```
   |
20 |   patchPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/servers/web-apps/davical/default.nix:24:3:
```
   |
24 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: nixos/tests/davical.nix
Comment (evils) [inline nixos/tests/davical.nix:2]: looks like you missed something here :)

File: maintainers/maintainer-list.nix
Comment (adenseaman) [reply maintainers/maintainer-list.nix:3954]: thank you, I split it into a separate commit

File: nixos/tests/davical.nix
Comment (adenseaman) [reply nixos/tests/davical.nix]: good catch! I've adjusted it

**Comment by ryantrinkle** [comment]
This would be great to have!

File: nixos/modules/services/web-apps/davical.nix
Comment (aanderse) [inline nixos/modules/services/web-apps/davical.nix:149]: Are you sure you want to automatically run database migrations for the sysadmin? It is generally seen as safer to allow/force the sysadmin to do this manually... Your call.

**Review by aanderse — CHANGES_REQUESTED** [review]
Hello! I have left some review points on your module. I hope you find them useful. If you have any questions please do not hesitate to ask for clarification. If this seems like an overwhelming amount of comments, please don't hesitate to ask for help - I'd be happy to chip in.

File: nixos/tests/davical.nix
Comment (symphorien) [inline nixos/tests/davical.nix:23]:
```suggestion
    page = machine.succeed("curl --fail -L http://localhost/")
```

File: pkgs/servers/web-apps/davical/default.nix
Comment (symphorien) [inline pkgs/servers/web-apps/davical/default.nix:4]: It is better to expose awl in all-packages.nix so that the end user can override it in an overlay.

**Comment by AndersonTorres** [comment]
Closing as dead.

**Pull request closed without merging**
